### DISCUSSION TITLE
Builtin function for setting a nested object key?

### DISCUSSION THREAD

**Discussion by caulagi** [open-policy-agent/community#231] [OPA and Rego] [answered]
Hi 👋 - How can I set a specific key for a nested object? For example, if I have - 

```
base_test_input := {"resource_changes": [{
	"address": "google_compute_instance.default",
	"type": "google_compute_instance",
	"change": {
		"actions": ["create"],
		"after": {"metadata": {"foo": "bar"}},
	},
}]}
```

How can 'override' only the `metadata` portion to different values for different tests? I have tried `object.union` but it seems to require copying most of the rest of the object structure that I would like to avoid copying, if possible.

I was wondering if something like `object.set` or `object.extent` would make sense`

**Comment by anderseknert** [comment, marked as answer]
You could use the `json.patch` [builtin function](https://www.openpolicyagent.org/docs/latest/policy-reference/#builtin-object-jsonpatch):
```rego
updated_meta := json.patch(base_test_input, [
	{"op": "replace", "path": "/resource_changes/0/change/after/metadata/foo", "value": "baz"}
])
```

**Reply by caulagi** [reply]
Thanks! I did look at `json.patch` but missed that I could index the array like you did.
